contracts/Turnstone.vy

```diff
@@ -0,0 +1,105 @@
+# @version 0.3.3
+"""
+@title Turnstone
+@author Volume.Finance
+"""
+
+MAX_VALIDATORS: constant(uint256) = 128
+MAX_BATCH: constant(uint256) = 128
+POWER_THRESHOLD: constant(uint256) = 2_863_311_530 # 2/3 of 2^32
+TURNSTONE_ID: immutable(bytes32)
+
+struct ValsetArgs:
+    validators: DynArray[address, MAX_VALIDATORS]
+    powers: DynArray[uint256, MAX_VALIDATORS]
+    nonce: uint256
```

Comment:
Based on usage this is not a `nonce`. We should use "trasaction id" or "message id" or something similar

Reply:
nonce is different than transaction id or message id. It is a number that increases every time to update valset. It is to make checkpoint different even the same valset later.

Reply:
Could we rename this to "valset_id" instead? Semantically we would be using this as nonce within the chain.